utils/roadie-backstage-entity-validator/src/schemas/annotations.schema.json

```diff
@@ -47,7 +47,7 @@
               "properties": {
                 "backstage.io/source-location": {
                   "type": "string",
-                  "pattern": "((url|gitlab|github|azure/api):.*|(dir):.*/)$"
+                  "pattern": "((url|gitlab|github|azure/api):(.*[^/]\\.[^/]*|[^.]*)/|(dir):.*/)$"
```

Comment:
Do these need to be capturing groups? Are the groups used somehow? If not, you could make them non-capturing. Also it's weird that `(dir)` is grouped with nothing.

Reply:
Thanks yes this wasn't quite working actually - it needed a much more complex solution apparently :(